contrib/babelfishpg_tsql/src/fts.c

```diff
@@ -64,3 +68,204 @@ babelfish_fts_rewrite(PG_FUNCTION_ARGS)
+        }
+
+        /* 
+         * Removing multiple spaces, tabs, newlines and double quotes from within the search string
+         * If any of the space or quote is encountered, we remove all their next occurances
+         * before end of the input or if next word is encountered and
+         * we append a single space instead.
+         * Case 1: '"word1   word2"' = 'word1 word2'
+         * Case 2: '"word1  "  word2 "' = 'word1 word2'
+         * Case 3: '"word1' + CHAR(9) + '"' = 'word1'
+         * Case 3: '"word1' + CHAR(10) + '" word2' = 'word1 word2'
+         * Case 4: '"word1  ' + CHAR(10) + ' ' + CHAR(10) + '"" = 'word1'
+         * Case 5: '"word1  ' + CHAR(9) + ' ' + CHAR(10) + ' ""  word2   "' = 'word1 word2'
+         */
+        else if ((left_ptr <= right_ptr) && (*left_ptr == ' ' || *left_ptr == '\t' || *left_ptr == '\n' || *left_ptr == '"'))
+        {
+            while ((left_ptr <= right_ptr) && (*left_ptr == ' ' || *left_ptr == '\t' || *left_ptr == '\n' || *left_ptr == '"'))
+            {
+                left_ptr++;
+            }
+            
+            if(left_ptr <= right_ptr && strlen(normalized_str.data) != 0)
+            {
+                appendStringInfoChar(&normalized_str, ' ');
+            }
+        }
+
+        else
+        {
+            appendStringInfoChar(&normalized_str, *left_ptr);
+            left_ptr++;
+        }
+    }
+
+    trimmed_input_str = normalized_str.data;
+    
+    if (!pltsql_allow_fulltext_parser)
+    {
+        ereport(ERROR,
+            (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+            errmsg("Full Text Search is not yet supported.")));
+    }
+        
+    /* 
+     * The normalized search string with text separated by single spaces are translated
+     * If a space is encountered it is converted to a bitwise OR operator.
+     * Case 1: 'word1' = 'word1'
+     * Case 2: 'word1 word2' = 'word1 | word2'
+     * Case 3: 'word1 word2 word3' = 'word1 | word2 | word3'
+     */
+    while(trimmed_input_str) 
+    {
+        char *ws;
+        ws = strchr(trimmed_input_str, ' ');
+        if(ws)
+        {
+            *ws = '\0';
+        }
+        appendStringInfo(&translated_str, "%s", trimmed_input_str);
+        trimmed_input_str = (ws) ? (ws+1) : NULL ;
+        if(trimmed_input_str != NULL)
+        appendStringInfoString(&translated_str, " | ");
+    }
+    translated_query = translated_str.data;
+    
+    if (translated_query) 
+    {
+        result_text = cstring_to_text(translated_query);
+    }
+
+    /*
+     * Make sure to free allocated memory.
+    */
+    pfree(input_str);
+    if(trimmed_input_str)
+    {
+        pfree(trimmed_input_str);
+    }
+    resetStringInfo(&normalized_str);
+
+    if (result_text) 
+    {
+        PG_RETURN_TEXT_P(result_text);
+    } 
+    else 
+    {
+        PG_RETURN_NULL();
+    }
+}
+
+static char 
+*trim(char *s, bool insideQuotes) {
```

Comment:
This function is similar to the one we are using in `fts_parser.y`. We should create a common utility function for this.